projects/igniteui-angular/src/lib/grids/selection/selection.service.ts

```diff
@@ -174,98 +180,139 @@ export class IgxGridCRUDService {
-            this.grid.onCellEditCancel.emit(args);
-            if (args.cancel) {
-                return;
-            }
+        if (this.cellInEditMode || this.rowInEditMode) {
             this.grid.endEdit(false);
             if (isEdge()) { this.grid.cdr.detectChanges(); }
             this.grid.tbody.nativeElement.focus();
         }
     }
 
-    isInEditMode(rowIndex: number, columnIndex: number): boolean {
+    public isInEditMode(rowIndex: number, columnIndex: number): boolean {
         if (!this.cell) {
             return false;
         }
         return this.cell.column.index === columnIndex && this.cell.rowIndex === rowIndex;
     }
+
+    public exitCellEdit(): boolean {
+        if (!this.cell) {
+            return false;
+        }
+
+        const args = this.cell?.createEditEventArgs(true);
+        this.grid.cellEditExit.emit(args);
+        if (args && args.cancel) {
+            return this._rowEditingBlocked = this._cellEditingBlocked = true;
+        }
+
+        this._rowEditingBlocked = this._cellEditingBlocked = false;
+        this.cell = null;
+        return false;
+    }
+
+    public exitRowEdit() {
+        this.row = null;
+    }
+
+    /** This method serves for unblocking row/cell edit when filter/sort is performed */
+    public releaseBlockedEditing() {
+        this._rowEditingBlocked = this._cellEditingBlocked = false;
+    }
+
+    /** This method is used when filtering/sortinging is triggered */
+    public exitEditRegardlessCancelation() {
+        this.cell = null;
+        if (this.grid.rowEditable) {
+            this.exitRowEdit();
+            this.grid.closeRowEditingOverlay();
+        }
+    }
+
+    private beginRowEdit(newCell) {
+        if (this.row && !this.sameRow(newCell.id.rowID)) {
+            this._rowEditingBlocked = this.grid.endEdit(true);
+            this.cell = newCell;
+            if (this.rowEditingBlocked) {
+                return true;
+            }
+
+            this._rowEditingBlocked = false;
+            this.exitRowEdit();
+        }
+
+        if (this.grid.rowEditable && (this.grid.primaryKey === undefined || this.grid.primaryKey === null)) {
+            console.warn('The grid must have a `primaryKey` specified when using `rowEditable`!');
+        }
+
+        if (!this.row) {
+            this.cell = newCell;
+            this.row = this.createRow(this.cell);
+            const rowArgs = this.row.createEditEventArgs(false);
+
+            this.grid.rowEditEnter.emit(rowArgs);
+            if (rowArgs.cancel) {
+                this.exitRowEdit();
+                this.cell = null;
+                return true;
+            }
+
+            this.row.transactionState = this.grid.transactions.getAggregatedValue(this.row.id, true);
+            this.grid.transactions.startPending();
+            this.grid.openRowOverlay(this.row.id);
+        }
+
+        this.emitCellEditEnter(newCell);
+    }
+
+    private beginCellEdit(cell) {
```

Comment:
That's still.. basically just `createCell`, doesn't begin anything. Not sure why the cell object has to carry the primary key and I'm not sure it's referenced anywhere, but it seems to be assigned all the time, so there's little reason for this method to exist.